rust/common-ifc/src/graph/mod.rs

```diff
@@ -0,0 +1,15 @@
+//! Abstract graph utilities for interacting with a
+//! recipe graph.
+
+mod port_graph;
+mod render;
+pub(crate) mod validation;
+
+pub use port_graph::*;
+pub use render::*;
```

Comment:
Is it possible to include `render` conditionally? It seems like this may be useful circumstantially, but a likely target to be cut for optimized builds.

Reply:
:100: 

I can imagine some future debugging flow where `InvalidGraph` errors have more distinction, containing a node/port that's invalid -- and can be used to debug graph construction by an author/LLM, either completely on the frontend, or rendered via `render_with_error(&graph, &error)`, which highlights the culprit components